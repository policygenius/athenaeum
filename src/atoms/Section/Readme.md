### Section Example:

```example
    <Section>
      <strong>New Component Created!</strong><br /><small>Now go and write some code! -- <em>You code monkey, you ;)</em></small>.
    </Section>
```
