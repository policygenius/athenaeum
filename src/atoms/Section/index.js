import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import colors from 'atoms/Color/colors.scss';
import styles from './section.module.scss';

function Section(props) {
  const {
    children,
    className,
    backgroundImageUrl,
    backgroundColor,
    color,
    spacing,
    center,
    padding,
    style
  } = props;

  const classes = [
    styles['section'],
    color && colors[color],
    colors[`background-${backgroundColor}`],
    spacing && styles[`top-${spacing}`],
    spacing && styles[`bot-${spacing}`],
    center && styles['center'],
    padding && styles['padding'],
    className
  ];

  return (
    <div
      className={classnames(...classes)}
      style={{
        ...style,
        backgroundImage: `url(${backgroundImageUrl})`
      }}
    >
      <div className={styles.wrapper}>
        { children }
      </div>
    </div>
  );
}


Section.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
  /**
   * Full string to be used with `url()` for the background-image
   */
  backgroundImageUrl: PropTypes.string,
  /**
   * Color style, see [Colors](#colors) for appropriate values
   */
  color: PropTypes.string,
  /**
   * Color style, see [Colors](#colors) for appropriate values
   */
  backgroundColor: PropTypes.string,
  /**
   * Spacer style, see [Spacers](#spacers) for appropriate values
   */
  spacing: PropTypes.number,
  /**
   * Center content
   */
  center: PropTypes.bool,
  /**
   * Padding, see [Spacers](#spacers) for appropriate values
   */
  padding: PropTypes.number,
  /**
   * React Style object
   */
  style: PropTypes.object,
};

Section.defaultProps = {
  // Place any default props here.
};

export default Section;
