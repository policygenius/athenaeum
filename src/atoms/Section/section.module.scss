 $base: .25;

.section {
  width: 100%;

  .wrapper {
    max-width: $page-max-width;
    margin: auto;
  }

  &.fullwidth {
    .wrapper {
      max-width: 100%;
    }
  }

  &.center {
    text-align: center;
  }
}

@for $i from 1 through 10 {
  .top-#{$i} { padding-top: ru($base * $i); }
  .bot-#{$i} { padding-bottom: ru($base * $i); }
}

.top-0 { padding-top: 0; }
.bot-0 { padding-bottom: 0; }

.top-11 { padding-top: ru(3); }
.bot-11 { padding-bottom: ru(3); }

.top-12 { padding-top: ru(6); }
.bot-12 { padding-bottom: ru(6); }

.padding {
  padding-left: ru(8);
  padding-right: ru(8);
}
