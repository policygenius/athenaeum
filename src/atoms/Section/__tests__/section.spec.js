import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Section from 'atoms/Section';

describe('<Section />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Section />);

    expect(wrapper.type()).toEqual(Section);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<Section />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
