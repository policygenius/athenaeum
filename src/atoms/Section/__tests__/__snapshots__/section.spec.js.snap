// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders correctly 1`] = `
<div
  className="section background-undefined"
  style={
    Object {
      "backgroundImage": undefined,
    }
  }
>
  <div
    className="wrapper"
  />
</div>
`;
